pkg/validate/lambda_validator.go

```diff
@@ -0,0 +1,91 @@
+package validate
+
+import (
+	"fmt"
+	"github.com/armory/spinnaker-operator/pkg/apis/spinnaker/interfaces"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"os"
+	"strconv"
+)
+
+const (
+	lambdaAccountType        		= "lambda"
+	lambdaClouddriverEnabledKey 	= "aws.features.lambda.enabled"
+	AccessKeyId 					= "providers.aws.accessKeyId"
+	SecretAccessKey 				= "providers.aws.secretAccessKey"
+)
+
+type lambdaValidator struct{}
+
+func (d *lambdaValidator) Validate(spinSvc interfaces.SpinnakerService, options Options) ValidationResult {
+
+	//AWS
+	//Check if the AWS Provider is enabled
+	awsAccountEnabled, err := spinSvc.GetSpinnakerConfig().GetHalConfigPropBool(awsAccountsEnabledKey, false)
+	if err != nil {
+		return ValidationResult{}
+	}
+
+	if !spinSvc.GetSpinnakerValidation().IsProviderValidationEnabled(awsAccountType) || !awsAccountEnabled {
+		return ValidationResult{}
+	}
+
+	//Get the AccessKeyId
+	awsAccessKey, err := spinSvc.GetSpinnakerConfig().GetHalConfigPropString(options.Ctx, AccessKeyId)
+	if err != nil {
+		return ValidationResult{Errors: []error{fmt.Errorf("AccessKeyId is missing")}}
+	}
+
+	//Get the SecretAccessKey
+	awsSecretKey, err := spinSvc.GetSpinnakerConfig().GetHalConfigPropString(options.Ctx, SecretAccessKey)
+	if err != nil {
+		return ValidationResult{Errors: []error{fmt.Errorf("SecretAccessKey is missing")}}
+	}
+
+	//Lambda
+	isEnabled, err := spinSvc.GetSpinnakerConfig().GetRawServiceConfigPropString("clouddriver",lambdaClouddriverEnabledKey)
+	if err != nil {
+		return ValidationResult{}
+	}
+	accountEnabled, err := strconv.ParseBool(isEnabled)
```

Comment:
Looks like this `err` is ignored, is that on purpose?

Reply:
The purpose of the err only happen if the property **aws.features.lambda.enabled** is missing: that means it is not enabled and then return no errors